tools/ci-build.sh

```diff
@@ -95,15 +93,18 @@ P4C_PIP_PACKAGES="ipaddr \
                   clang-format>=15.0.4"
 
 
+apt update
+
 if [[ "${DISTRIB_RELEASE}" == "18.04" ]] ; then
   P4C_DEPS+=" libprotobuf-dev protobuf-compiler"
 else
+  apt install -y curl gnupg
   echo "deb http://download.opensuse.org/repositories/home:/p4lang/xUbuntu_${DISTRIB_RELEASE}/ /" | tee /etc/apt/sources.list.d/home:p4lang.list
   curl -L "http://download.opensuse.org/repositories/home:/p4lang/xUbuntu_${DISTRIB_RELEASE}/Release.key" | apt-key add -
+  apt update
   P4C_DEPS+=" p4lang-bmv2"
 fi
 
-apt update
```

Comment:
I think we can keep a single `apt-get update` here as it will be executed after the "if" condition above.